### PR TITLE
2022-07-15 The permissions lifecycle for platform tools will be managed via version control

### PR DESCRIPTION
**Pull request by kjduensing**
## RFC Title
2022-07-15 - The permissions lifecycle for platform tools will be managed via version control
## RFC Preview Link
[Preview Link](https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/blob/amt/rfc/permissions-lifecycle/rfc/2022/2022-06-15%20-%20The%20permissions%20lifecycle%20for%20platform%20tools%20wiill%20be%20managed%20via%20version%20control.md)
## Comment Period End Date
07/15/2022

### CODE REVIEW

File: rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            amt_tf_review --> suggests2{Suggestions?}
+            suggests2 --> |No| merge[AMT merge]
+            suggests2 --> |Yes| pr
+            merge --> tf_apply
+      end
+    end
+```
+
+## Risks
+Adding permissions in plain text (JSON) format to a repo has some security implications. A malicious actor could see and potentially do harm with role, resource, and scope information if they were given access to the Github organization.
```
Comment (calquisola) [inline rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:99]: Maybe not needed, can we at least obfuscate the json data here ? E.g. maybe use something like base64 encoding.

Comment (kjduensing) [reply rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:99]: Good ideas @calquisola & @jchapin. One more idea is that we could potentially store the permissions files in the `va.gov-team-sensitive` repo. Though I've never done that before, so I'm not sure if that repo is intended for that kind of info. @td-usds might know.

Comment (mchelen-gov) [reply rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:99]: the `devops` repo is also private and contains IAM config

File: rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  init(0: AuthZ desired) --> impl(1: Describe permissions)
+  impl --> apply(2: Apply permissions)
+  apply --> use(3: Use permissions)
+  use --> revise(4: Revise permissions)
+  revise --> impl
+```
+
+### 0: Authorization Desired 
+This step handles the cases:
+1. The application has not been integrated into Keycloak
```
Comment (mchelen-gov) [inline rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:30]: how are applications identified within keycloak?

Comment (kjduensing) [reply rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:30]: info is potentially available w/in console. what's the integration lift?

File: rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+2. The application has been integrated into Keycloak, but has not configured authorization
+
+### 1: Describe Permissions
+This step encompasses the process of detailing a platform tool's permissions in JSON format<sub>[[RFC6]](https://vfs.atlassian.net/wiki/spaces/AMT/pages/2142011842/RFC+6+AMT+integrators+can+describe+fine-grained+permissions+using+JSON)</sub>, including any support from AMT.
+
+### 2: Apply Permissions
+The permissions application step includes the source control integration process. It also includes the automation required to apply the permissions configuration to the platform tool previously integrated into Keycloak. This is described in more detail below
```
Comment (mchelen-gov) [inline rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md:37]: how is this step kicked off, is it a manual process? how does a reviewer confirm the validity of the submitter's change?

**Comment by lhelmer** [comment]
Comment period complete @kjduensing

**Review by mchelen-gov — APPROVED** [review]
All comments are marked resolved.

**Comment by little-oddball** [comment]
Looked through this and I believe all comments were accounted for. I don't have the ability to close this.

**Pull request merged**